### PR TITLE
Fixes a version parsing bug and add missing macOS path

### PR DESCRIPTION
**Pull request by louislang**

### CODE REVIEW

File: pip/main.ts [hunk 1]
```diff
@@ -137,10 +162,15 @@ function parseDryRun(output: string): [object] {
   // Parse dependency names and versions.
   const deps = new_deps.split(' ');
   for (var i = 0; i < deps.length; i++) {
+    const pkg = deps[i];
```
Comment (louislang) [inline pip/main.ts:165]: `charset-normalizer-<ver>` caused issues with the previous logic as it split on the _first_ `-`. We need to split on the _last_ dash.

File: pip/main.ts [hunk 2]
```diff
@@ -52,13 +52,36 @@ const installStatus = PhylumApi.runSandboxed({
...
+async function logIssues(jobStatus) {
+  for(const key in jobStatus.packages) {
+    const pkg = jobStatus.packages[key];
+    const issues = pkg.issues;
+
+    for(const idx in issues) {
+      const issue = issues[idx];
+      let severity = issue.severity.toLowerCase();
+      
+      if(["high", "critical"].indexOf(severity) != -1) {
```
Comment (cd-work) [inline pip/main.ts:72]: JavaScript's `contains` method is `includes` iirc. I always get this wrong myself and it's a horrible name, but it should work here.

File: pip/main.ts [hunk 3]
```diff
@@ -110,6 +134,8 @@ async function checkDryRun() {
     console.error(
       `[${red("phylum")}] The operation caused a threshold failure.\n`,
     );
+
+    logIssues(jobStatus);
```
Comment (cd-work) [inline pip/main.ts:138]: Why was this included in this patch? To me it seems like an unrelated feature which should go into a different PR.

**Review by cd-work — CHANGES_REQUESTED** [review]
Looks good, would just split the changes into smaller PRs or at the very least independently mergeable commits.

File: pip/main.ts (hunk 3 above)
Comment (louislang) [reply pip/main.ts]: see: https://github.com/phylum-dev/community-extensions/pull/24

**Review by maxrake — APPROVED** [review]
Not much to add beyond @cd-work ...but I can help to test this locally and be the 2nd reviewer.
LGTM

File: pip/main.ts [hunk 4]
```diff
@@ -73,13 +73,15 @@ async function checkDryRun() {
       net: true,
     },
     stdout: 'piped',
+    stderr: 'piped',
   });
 
   // Ensure dry-run was successful.
   if (!status.success) {
+    console.log(status);
```
Comment (cd-work) [inline pip/main.ts:81]: Is this log here intentional? Seems like leftover debug information to me.

**Review by cd-work — CHANGES_REQUESTED** [review]

File: pip/main.ts (hunk 4 above)
Comment (louislang) [reply pip/main.ts]: debug stuff, removed!

File: pip/main.ts [hunk 5]
```diff
@@ -137,10 +163,15 @@ function parseDryRun(output: string): [object] {
   // Parse dependency names and versions.
   const deps = new_deps.split(' ');
   for (var i = 0; i < deps.length; i++) {
+    const pkg = deps[i];
+    const lastDashIndex = pkg.lastIndexOf('-');
+    const pkgName = pkg.substring(0, lastDashIndex);
+    const pkgVer = pkg.substring(lastDashIndex+1);
+
     const dep = deps[i].split('-');
```
[the remark this answers is not in the document]
Comment (louislang) [reply pip/main.ts]: good catch, removed!

**Review by cd-work — APPROVED** [review]

**Review by maxrake — APPROVED** [review]
LGTM
